src/Microsoft.Azure.WebJobs.Host/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    // Provides a blob-leased based implementation 
+    internal class BlobLeaseDistributedLockManager : IDistributedLockManager
+    {
+        internal const string FunctionInstanceMetadataKey = "FunctionInstance";
```

Comment:
>internal const string FunctionInstanceMetadataKey = "FunctionInstance"; [](start = 8, length = 71)

Might be a good idea to keep the IDistributedLockManager implementations free of Functions-specific things like this one.